### PR TITLE
GEODE-2828: AEQ created before the user region

### PR DESCRIPTION
**Pull request by nabarunnag**
Potential reviewers
@gesterzhou @upthewaterspout @jhuynh1 @ladyVader

### CODE REVIEW

**Comment by nabarunnag** [comment]
Pushed to develop
Failure in Travis fixed in GEODE-2632

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneEventListener.java [hunk 1]
```diff
@@ -53,6 +54,8 @@
   Logger logger = LogService.getLogger();
 
   private final RepositoryManager repositoryManager;
+  private CountDownLatch isFileAndChunkRegionReady = new CountDownLatch(1);
+  private boolean ready = false;
```
Comment (upthewaterspout) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneEventListener.java]: Does this need to be a separate flag? Seems like the CoutDownLatch already has the flag. If not this should be volatile.

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java [hunk 2]
```diff
@@ -166,28 +166,28 @@ public void createIndex(final String indexName, String regionPath, final Analyze
...
     if (this.managementListener != null) {
       this.managementListener.afterIndexCreated(index);
     }
+
+  }
+
+  public LuceneIndexImpl beforeDataRegionCreated(final String indexName, final String regionPath,
+      RegionAttributes attributes, final Analyzer analyzer,
+      final Map<String, Analyzer> fieldAnalyzers, String aeqId, final String... fields) {
+    LuceneIndexImpl index = createIndexRegions(indexName, regionPath);
```
Comment (jhuynh1) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java]: This wasn't introduced with this diff, but we probably want to rename this method.  I don't think the method is creating any index regions.  I think it's just creating the index object but none of the file/chunk or aeq region or buckets

Comment (nabarunnag) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneServiceImpl.java]: Changed it to createIndexObject - any better suggestion is welcomed

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneEventListener.java (hunk 1 above)
Comment (nabarunnag) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneEventListener.java]: Removed the flag and lucene precheckin was success.
Futher testing is ongoing

**Comment by nabarunnag** [comment]
Sending in a fresh pull request with all modification

**Pull request closed without merging**